gemini/node.py

```diff
@@ -30,4 +30,25 @@ def join_master() -> None:
 
     response = r.json()
 
-    print(response['node_id'] + "\n" + response['password'])
\ No newline at end of file
+    print(response['node_id'] + "\n" + response['password'])
+
+    auth_node(response['node_id'], response['password'])
+
+
+def auth_node(node_id, password) -> None:
+    print('authenticating node')
+    
+    url = 'http://localhost:5000/v1/nodes/auth'
+    headers = {'Content-Type': 'application/json'}
+
+    auth = {
+        'node_id': node_id,
+        'password': password,
+    }
+
+    r = requests.post(url, headers=headers, json=auth)
+    token = r.json()
```

Comment:
this is kind of misnamed: the json isn't the token, the token is the field inside the json.

also, i realize that there's more work here. you also have to check to see if the authentication was invalid (the credentials were wrong or something), this only covers the happy path